tests/_test_mscolab/test_server.py

```diff
@@ -233,7 +234,7 @@ def test_get_all_changes(self):
         with self.app.test_client() as test_client:
             operation, token = self._create_operation(test_client, self.userdata)
             fm, user = self._save_content(operation, self.userdata)
-            fm.save_file(operation.id, "content2", user)
+            fm.save_file(operation.id, XML_CONTENT2, user)
```

Comment:
My understanding is that the tests in this file should exercise the server's API and so should _not_ interact with MSColab internals like a FileManager instance. Instead, this should use the same code path that is also used by the MSUI client (so, socketio, I think).

Reply:
If we have other tests that give the same result, with the same coverage, we can reduce the internal tests you mentioned

Reply:
This case here is different. This is not an internal test. This is a test that exercises the external interface to interact with an operation, but in doing so erroneously uses internal structures.

Instead of calling fm.save_file this should go through a socketio test_client and emit the file-save event, so that it actually tests the server.

Reply:
not thought on that, indeed that should now with the improved test setup doable.

I thought on just fixing the wrong string...